File: js/anagram2.js
function anagramsFor(word, listOfWords) {
  let output = [];
  for (let m = 0; m < listOfWords.length; ++m) {
    let raw1 = listOfWords[m].toLowerCase().split(" ").join("");
    let raw2 = word.toLowerCase().split(" ").join("");
    let list1 = [];
    let list2 = [];
    for (let i = 0; i < raw1.length; ++i) {
      list1.push(raw1[i]);
    }
    for (let j = 0; j < raw2.length; ++j) {
      list2.push(raw2[j]);
    }
    list1.sort();
    list2.sort();
    if (list1.length == list2.length) {
      let counter = 0;
      for (let k = 0; k < list1.length; ++k) {
        if (list1[k] == list2[k]) {
          ++counter;
        }
      }
      if (counter == list1.length) {
        output.push(listOfWords[m]);
      }
    }
  }
  return output;
};

module.exports = anagramsFor

File: python/anagram2.py
# Don't forget to run the tests (and create some of your own)

def anagrams_for(word, list_of_words):
	output = []
	for element in list_of_words:
		raw1 = element.lower().replace(" ", "")
		raw2 = word.lower().replace(" ", "")
		list1 = []
		list2 = []
		for letter in raw1:
			list1.append(letter)
		for letter in raw2:
			list2.append(letter)
		if sorted(list1) == sorted(list2):
			output.append(element)
	return output
